pubsub/runtime/src/main/java/io/quarkiverse/googlecloudservices/pubsub/push/PubSubPushEndpointHandler.java

```diff
@@ -0,0 +1,133 @@
+package io.quarkiverse.googlecloudservices.pubsub.push;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Base64;
+import java.util.Map;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+import org.jboss.logging.Logger;
+
+import com.google.cloud.pubsub.v1.AckReplyConsumer;
+import com.google.cloud.pubsub.v1.MessageReceiver;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.util.Timestamps;
+import com.google.pubsub.v1.ProjectSubscriptionName;
+import com.google.pubsub.v1.PubsubMessage;
+
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Handler class for pubsub push messages.
+ */
+@ApplicationScoped
+public class PubSubPushEndpointHandler implements Handler<RoutingContext> {
+    private static final Logger LOGGER = Logger.getLogger(PubSubPushEndpointHandler.class.getName());
+
+    @Inject
+    PubSubPushManager manager;
+
+    /**
+     * Handler method for pubsub messages. This method is public in case you want to setup your own handler and augment
+     * that with additional logic. This handler method will be used in case the endpoint-path is configured.
+     *
+     * @param rc The routing context
+     */
+    @Override
+    public void handle(RoutingContext rc) {
+        var message = extractMessage(rc);
+        dispatchToListeners(rc, message);
+    }
+
+    private PubSubMessageJson extractMessage(RoutingContext rc) {
+        return rc.body().asPojo(PubSubMessageJson.class);
+    }
+
+    private void dispatchToListeners(RoutingContext rc, PubSubMessageJson message) {
+        var subscriptionName = ProjectSubscriptionName.parse(message.subscription);
+        LOGGER.debug("Received push message from subscription " + subscriptionName);
+
+        LOGGER.info("Received message at " + message.message().publishTime);
```

Comment:
I would be better to merge the two logs and use DEBUG

Reply:
Fixed, good catch, this was a bit of leftover debug logging which I missed removing. Fixed in latest commit.